### PR TITLE
added support for join Room out of a text message

### PR DESCRIPTION
**Pull request by syc0n**
We can now join a room by clicking on the the event ( #roomName) inside a message

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/CUModSquad/cu-patcher-ui/pull/74" target="_blank">ReviewNinja

File: src/sass/_chat.scss [hunk 1]
```diff
@@ -372,6 +372,11 @@
             margin-right: 0.5em;
             font-size: 80%;
           }
+          .chat-room-link {
+            color: $chat-line-room-color;
+            cursor: pointer;
+            margin-right: 0.5em;
```
Comment (Mehuge) [inline src/sass/_chat.scss]: This margin is not necessary, in fact it looks ugly. !fix

File: src/ts/content/chat/JoinRoomFromText.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import * as React from 'react';
+import * as events from '../../core/events';
+
+function fromText(text: string, keygen: () => number) : JSX.Element[] {
+  return [<span key={keygen()} className={'chat-room-link'} onClick={() : void => { events.fire('chat-show-room', text.substr(1)); }}>{text}</span>];
+}
+
+function createRegExp() : RegExp {
+  return /#[a-z0-9]+/g;
```
Comment (Mehuge) [inline src/ts/content/chat/JoinRoomFromText.tsx]: allow any non-whitespace as the room name. !fix

File: src/sass/_chat.scss (hunk 1 above)
Comment (syc0n) [inline src/sass/_chat.scss]: !fixed

File: src/ts/content/chat/JoinRoomFromText.tsx (hunk 2 above)
Comment (syc0n) [inline src/ts/content/chat/JoinRoomFromText.tsx]: !fixed

**Comment by Mehuge** [comment]
All looks good to me!

**Comment by syc0n** [comment]
resolved conflicts

**Pull request merged**